tests/unit_tests/Objects/variable/test_parameter.py

```diff
@@ -91,10 +113,326 @@ def test_init_value_max_exception(self):
                 url="url",
                 display_name="display_name",
                 callback=mock_callback,
-                enabled="enabled",
                 parent=None,
             )
 
+    def test_make_dependent_on(self, normal_parameter: Parameter):
+        # When
+        independent_parameter = Parameter(name="independent", value=1, unit="m", variance=0.01, min=0, max=10)
+        
+        # Then
+        normal_parameter.make_dependent_on(dependency_expression='2*a', dependency_map={'a': independent_parameter})
+
+        # Expect
+        assert normal_parameter._independent == False
+        assert normal_parameter.dependency_expression == '2*a'
+        assert normal_parameter.dependency_map == {'a': independent_parameter}
+        self.compare_parameters(normal_parameter, 2*independent_parameter)
+
+        # Then
+        independent_parameter.value = 2
+
+        # Expect
+        self.compare_parameters(normal_parameter, 2*independent_parameter)
+
+    def test_parameter_from_dependency(self, normal_parameter: Parameter):
+        # When Then
+        dependent_parameter = Parameter.from_dependency(
+            name = 'dependent', 
+            dependency_expression='2*a', 
+            dependency_map={'a': normal_parameter},
+            display_name='display_name',
+        )
+
+        # Expect
+        assert dependent_parameter._independent == False
+        assert dependent_parameter.dependency_expression == '2*a'
+        assert dependent_parameter.dependency_map == {'a': normal_parameter}
+        assert dependent_parameter.name == 'dependent'
+        assert dependent_parameter.display_name == 'display_name'
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+        # Then
+        normal_parameter.value = 2
+
+        # Expect
+        self.compare_parameters(dependent_parameter, 2*normal_parameter)
+
+    def test_dependent_parameter_with_unique_name(self, clear, normal_parameter: Parameter):
+        # When Then
+        dependent_parameter = Parameter.from_dependency(
+            name = 'dependent', 
+            dependency_expression='2*"Parameter_0"',
+        )
+
+        # Expect
+        assert dependent_parameter.dependency_expression == '2*"Parameter_0"'
```

Comment:
Now that I see this again, can you remind me why we can't simply do `2*Parameter_0"? First check if `Parameter_0` is defined in the map, and then check if it is a global parameter. 

It is not as unambiguous as using quotes to delimit the global parameters, but I'm afraid it would be unintuitive for users. I guess not many users would use the global_id anyway, so might be a nothing burger.

Reply:
Fair enough. Let's keep it as is.